third_party/radixlib/radixlib.c

```diff
@@ -237,21 +237,21 @@ radix_node_t* radix_matching_do(radix_tree_t* radix, prefix_t* prefix)
     if (node && node->prefix)
         stack[cnt++] = node;
 
-    while (cnt > 0)
-    {
-        node = stack[--cnt];
-        if (_comp_with_mask(PREFIX_TO_UCHAR(node->prefix),
-                            PREFIX_TO_UCHAR(prefix),
-                            node->prefix->bitlen))
-        {
-            return node;
+    node = NULL;
+    while (cnt-- > 0) {
```

Comment:
nit: I'd still assign `stack[cnt]` to a node just to make the code a bit cleaner.